### PR TITLE
Remove jackson-databind and jackson-annotations

### LINKED ISSUE #583: Remove jackson-databind and jackson-annotations dependencies now coming from core
jackson-databind and jackson-annotations were added as api dependencies in `server` of opensearch core.  Therfore, all plugins now have these as transitive dependencies.  Plugins that currently consume these dependencies directly will break while building unless the direct dependencies are removed.  If your plugin does not consume jackson-databind or jackson-annotations directly, feel free to close the related issue in your plugin's repository.

Example of solution in security plugin: https://github.com/opensearch-project/security/pull/2325

### PR DESCRIPTION
**Pull request by rupal-bq**
Signed-off-by: Rupal Mahajan <maharup@amazon.com>

### Description
Remove jackson-databind and jackson-annotations

Updated dompurify to 2.4.1 to resolve conflicts

### Issues Resolved
#583

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-reports/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "async-mutex": "^0.2.6",
     "babel-polyfill": "^6.26.0",
     "cron-validator": "^1.1.1",
-    "dompurify": "^2.3.8",
+    "dompurify": "^2.4.1",
```
Comment (ps48) [inline dashboards-reports/package.json:22]: Is this change intentional? don't find mention of this in PR description

**Review by joshuali925 — APPROVED** [review]

**Review by sejli — APPROVED** [review]

File: dashboards-reports/package.json (hunk 1 above)
Comment (rupal-bq) [reply dashboards-reports/package.json:22]: Yes it is intentional. Build was failing with error 
```ERROR [single_version_dependencies] Multiple version ranges for the same dependency
      were found declared across different package.json files. Please consolidate
      those to match across all package.json files. Different versions for the
      same dependency is not supported.

      If you have questions about this please reach out to the operations team.

      The conflicting dependencies are:

        dompurify
          ^2.4.1 => opensearch-dashboards
          ^2.3.8 => reports-dashboards
error Command failed with exit code 1.
```
Ref: https://github.com/opensearch-project/dashboards-reporting/actions/runs/3701723685/jobs/6271294102

Comment (rupal-bq) [reply dashboards-reports/package.json:22]: added in PR description
